tensorflow/core/kernels/resource_variable_ops.cc

```diff
@@ -1108,6 +1132,37 @@ REGISTER_KERNEL_BUILDER(Name("ResourceScatterUpdate")
 
 #endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
+#ifdef TENSORFLOW_USE_DIRECTML
+#define REGISTER_SCATTER_ARITHMETIC_DML(type) \
+  REGISTER_SCATTER_ARITHMETIC(type, CPU);
+#define REGISTER_SCATTER_MINMAX_DML(type) REGISTER_SCATTER_MINMAX(type, CPU);
+
+#define REGISTER_SCATTER_UPDATE_DML(type) REGISTER_SCATTER_UPDATE(type, CPU);
+
+TF_CALL_GPU_NUMBER_TYPES_NO_HALF(REGISTER_SCATTER_ARITHMETIC_DML);
+TF_CALL_GPU_NUMBER_TYPES_NO_HALF(REGISTER_SCATTER_MINMAX_DML);
```

Comment:
Like Gather, these should have their own DML-specific implementations since they are float types, not Variant.